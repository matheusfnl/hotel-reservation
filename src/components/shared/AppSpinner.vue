<template>
  <span class="pulse-animation full-width flex justify-center">
    <q-icon :color="color" :size="size" name="hotel" />
  </span>
</template>

<script lang="ts" setup>
defineProps<{
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
  color: string
}>()
</script>

<style lang="scss" scoped>
.pulse-animation {
  animation: pulse 1s ease-in-out infinite;
  transform-origin: center center;
}

@keyframes pulse {
  0% {
    opacity: 1;
    transform: scale(1);
    transform: scale(1.05) rotate(0deg);
  }
  25% {
    opacity: 0.3;
    transform: scale(1.05) rotate(5deg);
  }
  50% {
    opacity: 0.6;
    transform: scale(1.1);
    transform: scale(1.05) rotate(0deg);
  }
  75% {
    opacity: 0.9;
    transform: scale(1.05);
    transform: scale(1.05) rotate(-5deg);
  }
  100% {
    opacity: 1;
    transform: scale(1);
    transform: scale(1.05) rotate(0deg);
  }
}
</style>
