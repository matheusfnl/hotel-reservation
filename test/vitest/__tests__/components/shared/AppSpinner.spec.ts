import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import AppSpinner from '@/components/shared/AppSpinner.vue'

describe('AppSpinner', () => {
  it('should render correctly', () => {
    const wrapper = mount(AppSpinner)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
