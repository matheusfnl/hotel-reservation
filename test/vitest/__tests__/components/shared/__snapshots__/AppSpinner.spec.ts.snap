// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppSpinner > should render correctly 1`] = `"<span data-v-6df88b7b="" class="pulse-animation full-width flex justify-center"><i data-v-6df88b7b="" class="q-icon notranslate material-icons" aria-hidden="true" role="presentation">hotel</i></span>"`;
